File: AssignCookies/AssignCookies_455.go
package main

import (
	"fmt"
	"sort"
)

func assignCookies(g []int, s []int) int {
	sort.Ints(g)
	sort.Ints(s)
	child, cookie := 0, 0

	for child < len(g) && cookie < len(s) {
		if s[cookie] >= g[child] {
			child++
		}
		cookie++
	}
	return child
}

func main() {
	g := []int{1, 2, 3}
	s := []int{1, 1}
	fmt.Println(assignCookies(g, s))
}

File: LeetCodeSolves/AssignCookies_455.py
def assignCookies(g, s):
    g.sort()
    s.sort()
    child, cookie = 0, 0
    while child < len(g) and cookie < len(s):
        if s[cookie] >= g[child]:
            child += 1
        cookie += 1
    return child


if __name__ == '__main__':
    g = [1, 2, 3]
    s = [1, 1]
    print(assignCookies(g, s))